File: examples/nodejs/send-text.ts
/**
 * Zé da API — Enviar mensagem de texto
 * Documentação: https://api.zedaapi.com/docs
 *
 * Uso: npx tsx send-text.ts
 */

// Configuração — substitua pelos seus dados
const HOST = process.env.ZEDAAPI_HOST || "https://sua-instancia.zedaapi.com";
const INSTANCE_ID = process.env.ZEDAAPI_INSTANCE_ID || "sua-instancia";
const INSTANCE_TOKEN = process.env.ZEDAAPI_INSTANCE_TOKEN || "seu-token-aqui";

// URL base com autenticação embutida na rota
const BASE_URL = `${HOST}/instances/${INSTANCE_ID}/token/${INSTANCE_TOKEN}`;

interface SendTextResponse {
  key?: {
    remoteJid: string;
    fromMe: boolean;
    id: string;
  };
  message?: Record<string, unknown>;
  error?: string;
}

async function sendText(
  phone: string,
  message: string,
): Promise<SendTextResponse> {
  const response = await fetch(`${BASE_URL}/send-text`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ phone, message }),
  });

  if (!response.ok) {
    const error = await response.text();
    throw new Error(`Erro ${response.status}: ${error}`);
  }

  return response.json() as Promise<SendTextResponse>;
}

// Exemplo de uso
async function main() {
  try {
    const result = await sendText(
      "5511999999999",
      "Olá! Mensagem enviada via Node.js.",
    );
    console.log(
      "Mensagem enviada com sucesso:",
      JSON.stringify(result, null, 2),
    );
  } catch (error) {
    console.error("Falha ao enviar mensagem:", error);
    process.exit(1);
  }
}

main();

File: examples/python/send_text.py
"""
Zé da API — Enviar mensagem de texto
Documentação: https://api.zedaapi.com/docs

Uso: python send_text.py
Requisitos: pip install requests
"""

import os
import sys

import requests

# Configuração — substitua pelos seus dados
HOST = os.getenv("ZEDAAPI_HOST", "https://sua-instancia.zedaapi.com")
INSTANCE_ID = os.getenv("ZEDAAPI_INSTANCE_ID", "sua-instancia")
INSTANCE_TOKEN = os.getenv("ZEDAAPI_INSTANCE_TOKEN", "seu-token-aqui")

# URL base com autenticação embutida na rota
BASE_URL = f"{HOST}/instances/{INSTANCE_ID}/token/{INSTANCE_TOKEN}"


def send_text(phone: str, message: str) -> dict:
    """Envia uma mensagem de texto via Zé da API."""
    response = requests.post(
        f"{BASE_URL}/send-text",
        headers={"Content-Type": "application/json"},
        json={"phone": phone, "message": message},
        timeout=30,
    )
    response.raise_for_status()
    return response.json()


if __name__ == "__main__":
    try:
        result = send_text("5511999999999", "Olá! Mensagem enviada via Python.")
        print("Mensagem enviada com sucesso:")
        print(result)
    except requests.HTTPError as e:
        print(f"Erro HTTP: {e.response.status_code} - {e.response.text}")
        sys.exit(1)
    except requests.RequestException as e:
        print(f"Erro de conexão: {e}")
        sys.exit(1)
